### PR TITLE
Add tests to save and return feature

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

This PR adds a feature test that tests the save and continue behaviour.

The acceptance tests use the mocked submitter to retrieve the token that runner sent to submitter. Then by capturing the link, the test mimics what the user would do to click the email and recover the form data.

## Plus

This PR adds site prism gem which is a gem that enable us to create page objects. The benefits of this pattern can be seen here: https://martinfowler.com/bliki/PageObject.html
Plus the diff will show that the tests are more dry and more behavior-related rather than implementation-related details.

### CODE REVIEW

File: tests/spec/features/save_and_return_module_spec.rb [hunk 1]
```diff
@@ -1,107 +1,138 @@
 require 'capybara/rspec'
 require 'spec_helper'
 
+class SaveAndReturnApp < SitePrism::Page
```
Comment (sequencemedialimited) [inline tests/spec/features/save_and_return_module_spec.rb]: Brilliant

File: tests/spec/features/save_and_return_module_spec.rb [hunk 2]
```diff
@@ -1,107 +1,138 @@
...
 
   it 'puts a save and return message into the page' do
-    visit 'http://features-save-and-return-app:3000'
-    click_on 'Start'
-
-    expect(page).to have_selector 'p', text: /Your session will time out after \d+ minutes of inactivity/
-    expect(page).to have_selector 'p', text: 'Save your details if you want to come back later'
+    form.start_button.click
+    expect(form).to have_content(
+      /Your session will time out after \d+ minutes of inactivity/
```
Comment (sequencemedialimited) [inline tests/spec/features/save_and_return_module_spec.rb]: Do we need this? Shouldn't we test this in its own spec?

Comment (tomas-stefano) [reply tests/spec/features/save_and_return_module_spec.rb]: I don't know. This was already there, I just adapted. Should I delete then?

Comment (sequencemedialimited) [reply tests/spec/features/save_and_return_module_spec.rb]: Hm, I must have written it, then

Kill it

Comment (sequencemedialimited) [reply tests/spec/features/save_and_return_module_spec.rb]: Actually, I remember writing the `\d` part. You can kill it

**Pull request merged**